examples/3d/perspective_zoom.rs

```diff
@@ -0,0 +1,125 @@
+//! Shows how to zoom and orbit a perspective projection camera.
+
+use std::ops::Range;
+
+use bevy::{input::mouse::AccumulatedMouseScroll, prelude::*};
+
+#[derive(Debug, Default, Reflect, Resource)]
+struct CameraSettings {
+    // Multiply keyboard inputs by this factor
+    pub orbit_speed: f32,
+    // Clamp fixed vertical scale to this range
+    pub zoom_range: Range<f32>,
+    // Multiply mouse wheel movements by this factor
+    pub zoom_speed: f32,
+}
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<CameraSettings>()
+        .add_systems(Startup, (setup, instructions))
+        .add_systems(Update, camera_controls)
+        .register_type::<CameraSettings>()
+        .run();
+}
+
+/// Set up a simple 3D scene
+fn setup(
+    mut camera_settings: ResMut<CameraSettings>,
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    camera_settings.orbit_speed = 0.02;
+    // Clamp FOV to this range. Note that we can't adjust FOV to more than PI,
+    // which represents a 180 degree field.
+    camera_settings.zoom_range = 0.5..3.0;
+    camera_settings.zoom_speed = 1.0;
+
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(5.0, 5.0, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
+        ..default()
+    });
+
+    // Plane
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Plane3d::default().mesh().size(5.0, 5.0)),
+        material: materials.add(Color::srgb(0.3, 0.5, 0.3)),
+        ..default()
+    });
+
+    // Cube
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Cuboid::default()),
+        material: materials.add(Color::srgb(0.8, 0.7, 0.6)),
+        transform: Transform::from_xyz(1.5, 0.5, 1.5),
+        ..default()
+    });
+
+    // Light
+    commands.spawn(PointLightBundle {
+        transform: Transform::from_xyz(3.0, 8.0, 5.0),
+        ..default()
+    });
+}
+
+fn instructions(mut commands: Commands) {
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                align_items: AlignItems::Start,
+                flex_direction: FlexDirection::Column,
+                justify_content: JustifyContent::Start,
+                width: Val::Percent(100.),
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                "Scroll mouse wheel to zoom in/out",
+                TextStyle::default(),
+            ));
+            parent.spawn(TextBundle::from_section(
+                "A or D to orbit left or right",
+                TextStyle::default(),
+            ));
+        });
+}
+
+fn camera_controls(
```

Comment:
Any reason why this does not orbit on yaw *and* pitch? That would be more useful to most users imo.

Reply:
Because I'd only just figured out how to do the former :laughing: lemme take another look.

Reply:
Here's a reference: https://github.com/janhohenheim/avian_interpolation/blob/9cc03a508b91fde6e2eaa1add471c0505473d5c2/crates/avian_interpolation3d/examples/follow_object.rs#L125-L154
You can just combine these two systems into one.